From 44c12c4746c3b25285d0895813e0f969bf13522b Mon Sep 17 00:00:00 2001
From: Alexander Larsson <alexl@redhat.com>
Date: Fri, 12 Jan 2018 16:52:39 +0100
Subject: [PATCH] FcHashTableAddInternal: Compare against the right key

We were comparing the passed in key with the ready-to-insert key
rather than the key in the hashtable, so if you ever had a hash
conflicts we'll never insert the new item.

https://bugs.freedesktop.org/show_bug.cgi?id=101889
---
 src/fchash.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fchash.c b/src/fchash.c
index ba6a453..396f452 100644
--- a/src/fchash.c
+++ b/src/fchash.c
@@ -178,7 +178,7 @@ FcHashTableAddInternal (FcHashTable *table,
     for (prev = &table->buckets[hash % FC_HASH_SIZE];
 	 (b = fc_atomic_ptr_get (prev)); prev = &(b->next))
     {
-	if (!table->compare_func (bucket->key, key))
+	if (!table->compare_func (b->key, key))
 	{
 	    if (replace)
 	    {
-- 
2.13.6

